### ISSUE TITLE
Please improved it's "Now Playing" tab

### ISSUE THREAD

**Issue by iBabu007** [FossifyOrg/Music-Player#31]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Music-Player/issues) or [closed](https://github.com/FossifyOrg/Music-Player/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Music-Player/discussions) or [closed](https://github.com/FossifyOrg/Music-Player/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Music-Player/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Please improved it's "Now Playing" tab.

### Why do you want this feature?

it's current "Now playing" tab is not handy.
You can't move sliding bar with single hand.
![Screenshot_20240218-191900_1_1](https://github.com/FossifyOrg/Music-Player/assets/65340361/35b7312d-868f-4d4c-bc26-6d2de423bbc0)

**Comment by Aga-C** [comment]
Can you elaborate a bit more, what would you like to be done? I understand that the basic thing is to move things on the UI so they are more accessible with one hand, but I see more things on the screenshot:

- Do you want to add Sleep Timer and Equalizer to the UI?
- Do you want to combine Equalizer with Speed into one dialog?
- "Paused after complete" is already under the repeat icon (tap multiple times to turn it on). Do you want to move it to another place?
- There are no favorites in the app. If you want that feature, please raise it separately.

**Comment by iBabu007** [comment]
If possible then add 3-4 buttons for any important task like (equalizer, share, info, favorite, speed, sleep timer, etc.)
Mainly sleep timer, equalizer, Paused after complete (if separated form loop button)

If "Paused after complete" button under loop button then it would be separated or add long tap loop button for "paused after completed"

**Comment by iBabu007** [comment]
> * Do you want to add Sleep Timer and Equalizer to the UI?
Yes
> * Do you want to combine Equalizer with Speed into one dialog?
No
> * "Paused after complete" is already under the repeat icon (tap multiple times to turn it on). Do you want to move it to another place?
It's current position is very confusing and if possible then add long press loop button.
> * There are no favorites in the app. If you want that feature, please raise it separately.
Ok

**Comment by HolonProduction** [comment]
I did a mockup for an improved playback screen since I don't really like the visuals of the current and thought I'd put it here:

![image](https://github.com/FossifyOrg/Music-Player/assets/99511741/1920525e-ab1b-4285-aa4b-d49b0ea480a2)

Notable changes:
- Consistent use of rounded material design icons
- The thumbnail now has a margin which allows to place a row of buttons above it (this allows for the equalizer to stay roughly in the same place)
- The three dots menu can be used to present many less used options for example the sleep timer, but also actions related to the current song (think playlist management)
- The time slider is side by side with playback related buttons like pause and play
- The playback speed options was moved to the top and the current speed is only shown when it was changed, also it isn't center aligned which prevents the icon from jumping around based on the length of the multiplier string (I also changed the icon to the speed icon from material design since I see it used more often in other apps so it should be more familiar in my opinion)
- For accessibility loop and shuffle show a dot beneath if active
- The space above the time line could be used to place more frequently used actions like add to playlist or favorite if that were to be implemented
- The preview to the next song has the queue icon next to it, to make clear that it can open the queue

Improvements not visible in the mockup:
- consistent use of tooltips
- take the "stop after current song" option out of the rotation for the loop button. It will still display in the same place if active, but it can only be toggled through the menu in the top right (were it would be placed near to the sleep timer since those functions have some similarities in the use case)
- remove the system snackbar message when toggling between loop options, tooltips should suffice
- for songs without thumbnails use a card like fallback that has a different color than the background, also replace the old simple mobile tools logo with the new fossify logo there
- also it could be a nice touch to adjust the material you color scheme on this screen based on the thumbnail, would need to see it in action though to be sure

I'd be willing to implement a rework of the playback screen, but I would appreciate some feedback first.

**Comment by user24024** [comment]
> The thumbnail now has a margin which allows to place a row of buttons above it

I think it would be better to have the album cover and track info at the very top, and everything else (including those additional buttons) as close to the bottom as possible, for better single-handed use. Even if swiping the album cover to change songs would be used by most people, it would still be easy to do, as the album cover is rather large, unlike the buttons :)

> for songs without thumbnails use a card like fallback that has a different color than the background, also replace the old simple mobile tools logo with the new fossify logo there

Sort of related - it would be great to have an option to disable the album cover field altogether, or at least make it scalable depending on available space. In any case, it would be great to have the album specified in text. #29 is also related to this.
Pitch and balance control are not implemented yet (I just raised the issues for those :) ), but if they are implemented at some point, I would love to have permanently visible sliders for balance, pitch and speed, instead of the album cover.

**Comment by HolonProduction** [comment]
> I think it would be better to have the album cover and track info at the very top, and everything else (including those additional buttons) as close to the bottom as possible, for better single-handed use.

The space is meant for actions that are not used often so I don't think single handed use is a big concern. And placing the overflow menu anywhere else will make it hard to find, since it is placed in that place across almost all fossify apps and also a lot of other android apps ([this is kind of standardized in the material design specs for top app bars](https://m3.material.io/components/top-app-bar/guidelines#224e8ca0-4177-45f4-9f4f-eef1fbb0e4b2)).

However I was thinking about removing the preview completely and moving the progress bar group to the very bottom, and placing the queue icon in the space above the bar (that's how Samsung music does it). But I didn't go with it, since removing elements always is a bit hard to get approval on.

> or at least make it scalable depending on available space

Not sure how that would look. Have a mockup or something?

> In any case, it would be great to have the album specified in text

Spotify shows this info in the top app bar. But to have the space we would need to move the equalizer and speed buttons into the overflow menu. Given that personally I never really looked at it in Spotify, I'm not sure if it is worth it.

> I would love to have permanently visible sliders for balance, pitch and speed, instead of the album cover.

I mean having a toggle in the settings for "DJ Mode" or whatever, which replaces the icon with playback controls is always a possibility, but that goes beyond the scope of this issue in my opinion.

**Comment by user24024** [comment]
> since it is placed in that place across almost all fossify apps

Yeah, I forgot that this isn't just a single app, but a part of a suite. I wouldn't like the design choice, but I'd understand the desire for a standardized look.


> Not sure how that would look. Have a mockup or something?

I've combined the screenshots from the main page:

![image](https://github.com/FossifyOrg/Music-Player/assets/160588372/cc0301d9-c6bc-4a8e-86df-00dd57648aa3)

Lots of space for an EQ and/or other extras :)


> having a toggle in the settings for "DJ Mode" or whatever, which replaces the icon with playback controls is always a possibility, but that goes beyond the scope of this issue in my opinion.

I guess it would be easier to have separate "themes" for the 'Now Playing' screen (like Metro does, for example) instead of a modular design, but it might be too far fetched still.

**Comment by NickAndreou** [comment]
Hello, I requested back in June the equaliser to be added to the play screen and my request was closed as a duplicate of this...rant. I would like to know when the equaliser will be added, that is all I want, just one button, thanks.

And while we are at it - add the equaliser on the lock screen here:

<img alt="ignoreImageMinify" src="https://github.com/user-attachments/assets/26284fb6-503f-404b-82d1-397db70a521f" width=180 />
